Update to Molecule 3

Current builds are failing due to the changes in lint configuration between Molecule 2 and 3

Builds fail with this message:
```
ERROR: Failed to pre-validate.

{'verifier': [{'name': ['unallowed value goss']}]}
Makefile:55: recipe for target 'test' failed
make: *** [test] Error 1
The command "make test" exited with 2.
```

--- a/.ansible-lint
+++ b/.ansible-lint
@@ -3,5 +3,3 @@ exclude_paths:
   - ./tasks/old_content
 x:
   - '303'
-  - '306'
-  - '602'

--- a/.ansible-lint
+++ b/.ansible-lint
@@ -1,0 +1,3 @@
+---
+exclude_paths:
+  - ./tasks/old_content

--- a/.ansible-lint
+++ b/.ansible-lint
@@ -1,3 +1,7 @@
 ---
 exclude_paths:
   - ./tasks/old_content
+x:
+  - '303'
+  - '306'
+  - '602'

--- a/Makefile
+++ b/Makefile
@@ -47,8 +47,7 @@ clean:
 
 .PHONY: lint
 lint: bin/python
-	( . bin/activate && find . -name "*.yml" |xargs bin/yamllint )
-	( . bin/activate && bin/molecule lit )
+	( . bin/activate && bin/molecule lint )
 
 .PHONY: test
 test: bin/python

--- a/molecule/default/molecule.yml
+++ b/molecule/default/molecule.yml
@@ -4,10 +4,10 @@ dependency:
   requirements_file: requirements.yml
 driver:
   name: docker
-verifier:
-  name: goss
-  lint:
-    name: yamllint
+lint: |
+  set -e
+  yamllint .
+  ansible-lint
 platforms:
   - name: CIS-centos7
     image: milcom/centos7-systemd
@@ -19,12 +19,6 @@ provisioner:
       remote_tmp: /tmp
   playbooks:
     converge: playbook.yml
-  lint:
-    name: ansible-lint
-    options:
-      exclude:
-        - tasks/old_content
-      x: ["303", "306", "602"]
 scenario:
   name: default
   create_sequence:

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,3 +1,5 @@
+ansible-lint
+yamllint
 tox
 molecule
 docker

--- a/tasks/prelim.yml
+++ b/tasks/prelim.yml
@@ -8,13 +8,13 @@
   check_mode: no
 
 - name: "PRELIM | Gather accounts with empty password fields"
-  shell: "cat /etc/shadow | awk -F: '($2 == \"\" ) {j++;print $1; } END {exit j}'"
+  command: "awk -F: '($2 == \"\" ) {j++;print $1; } END {exit j}' /etc/shadow"
   register: empty_password_accounts
   changed_when: no
   check_mode: no
 
 - name: "PRELIM | Gather UID 0 accounts other than root"
-  shell: "cat /etc/passwd | awk -F: '($3 == 0 && $1 != \"root\") {i++;print $1 } END {exit i}'"
+  command: "awk -F: '($3 == 0 && $1 != \"root\") {i++;print $1 } END {exit i}' /etc/passwd"
   register: uid_zero_accounts_except_root
   changed_when: no
   check_mode: no

--- a/tasks/section1.yml
+++ b/tasks/section1.yml
@@ -438,7 +438,7 @@
       - rule_1.1.20
 
 - name: "SCORED | 1.1.21 | PATCH | Ensure sticky bit is set on all world-writable directories"
-  shell: df --local -P | awk {'if (NR!=1) print $6'} | xargs -I '{}' find '{}' -xdev -type d -perm -0002 2>/dev/null | xargs chmod a+t
+  shell: "set -o pipefail && df --local -P | awk {'if (NR!=1) print $6'} | xargs -I '{}' find '{}' -xdev -type d -perm -0002 2>/dev/null | xargs chmod a+t"
   changed_when: no
   failed_when: no
   #when: sticky_bit_on_worldwritable_dirs_audit.rc == '0'
@@ -661,7 +661,7 @@
   notify: generate new grub config
   when:
       - rhel7cis_set_boot_pass|bool
-      - grub_pass is defined and grub_pass.passhash is defined and grub_pass.passhash != ''
+      - grub_pass is defined and grub_pass.passhash is defined and grub_pass.passhash
       - rhel7cis_rule_1_4_2|bool
   tags:
       - level1
@@ -733,7 +733,7 @@
       - rule_1.5.1
 
 - name: "NOTSCORED | 1.5.2 | PATCH | Ensure XD/NX support is enabled"
-  shell: dmesg|grep -E "NX|XD" | grep " active"
+  shell: 'set -o pipefail && dmesg | grep -E "NX|XD" | grep " active"'
   changed_when: no
   when:
       - rhel7cis_rule_1_5_2|bool

--- a/tasks/section1.yml
+++ b/tasks/section1.yml
@@ -733,7 +733,7 @@
       - rule_1.5.1
 
 - name: "NOTSCORED | 1.5.2 | PATCH | Ensure XD/NX support is enabled"
-  shell: 'set -o pipefail && dmesg | grep -E "NX|XD" | grep " active"'
+  shell: set -o pipefail && dmesg | grep -E "NX|XD" | grep " active"
   changed_when: no
   when:
       - rhel7cis_rule_1_5_2|bool

--- a/tasks/section1.yml
+++ b/tasks/section1.yml
@@ -438,7 +438,7 @@
       - rule_1.1.20
 
 - name: "SCORED | 1.1.21 | PATCH | Ensure sticky bit is set on all world-writable directories"
-  shell: "set -o pipefail && df --local -P | awk {'if (NR!=1) print $6'} | xargs -I '{}' find '{}' -xdev -type d -perm -0002 2>/dev/null | xargs chmod a+t"
+  shell: set -o pipefail && df --local -P | awk {'if (NR!=1) print $6'} | xargs -I '{}' find '{}' -xdev -type d -perm -0002 2>/dev/null | xargs chmod a+t
   changed_when: no
   failed_when: no
   #when: sticky_bit_on_worldwritable_dirs_audit.rc == '0'

--- a/tasks/section4.yml
+++ b/tasks/section4.yml
@@ -218,7 +218,10 @@
       - rule_4.1.11
 
 - name: "SCORED | 4.1.12 | PATCH | Ensure use of privileged commands is collected"
-  shell: for i in  $(df | grep '^/dev' | awk '{ print $NF }'); do find $i -xdev -type f -perm -4000 -o -type f -perm -2000 2>/dev/null; done
+  shell: "set -o pipefail && \
+    for i in $(df | grep '^/dev' | awk '{ print $NF }'); do \
+      find $i -xdev -type f -perm -4000 -o -type f -perm -2000 2>/dev/null; \
+    done"
   register: priv_procs
   changed_when: no
   check_mode: no

--- a/tasks/section4.yml
+++ b/tasks/section4.yml
@@ -218,10 +218,11 @@
       - rule_4.1.11
 
 - name: "SCORED | 4.1.12 | PATCH | Ensure use of privileged commands is collected"
-  shell: "set -o pipefail && \
-    for i in $(df | grep '^/dev' | awk '{ print $NF }'); do \
-      find $i -xdev -type f -perm -4000 -o -type f -perm -2000 2>/dev/null; \
-    done"
+  shell: >-
+    set -o pipefail &&
+      for i in $(df | grep '^/dev' | awk '{ print $NF }'); do
+        find $i -xdev -type f -perm -4000 -o -type f -perm -2000 2>/dev/null;
+    done
   register: priv_procs
   changed_when: no
   check_mode: no

--- a/tasks/section5.yml
+++ b/tasks/section5.yml
@@ -361,7 +361,7 @@
         notify:
             - restart sshd
         when:
-            - "rhel7cis_sshd['allowusers']|default('') != ''"
+            - rhel7cis_sshd['allowusers'] | default('')
 
       - name: "SCORED | 5.2.15 | PATCH | Ensure SSH access is limited - allowgroups"
         lineinfile:
@@ -372,7 +372,7 @@
         notify:
             - restart sshd
         when:
-            - "rhel7cis_sshd['allowgroups']|default('') != ''"
+            - rhel7cis_sshd['allowgroups'] | default('')
 
       - name: "SCORED | 5.2.15 | PATCH | Ensure SSH access is limited - denyusers"
         lineinfile:
@@ -383,7 +383,7 @@
         notify:
             - restart sshd
         when:
-            - "rhel7cis_sshd['denyusers']|default('') != ''"
+            - rhel7cis_sshd['denyusers'] | default('')
 
       - name: "SCORED | 5.2.15 | PATCH | Ensure SSH access is limited - denygroups"
         lineinfile:
@@ -394,7 +394,7 @@
         notify:
             - restart sshd
         when:
-            - "rhel7cis_sshd['denygroups']|default('') != ''"
+            - rhel7cis_sshd['denygroups'] | default('')
   when:
       - rhel7cis_rule_5_2_15|bool
   tags:

